src/marten/db/query/set.cr

```diff
@@ -704,6 +704,82 @@ module Marten
+        #
+        # This method allows retrieving a record based on a custom SQL condition without parameters.
+        # It returns `nil` if no record matches the condition.
+        #
+        # Example:
+        # ```
+        # tag = Tag.all.get("is_active = true")
+        # ```
+        def get(raw_predicate : String)
+          raise_empty_raw_predicate if raw_predicate.empty?
+          get(Node.new(raw_predicate))
+        end
+
+        # Returns the model instance matching the given raw SQL condition with positional arguments.
+        #
+        # This method allows retrieving a record based on a custom SQL condition with positional arguments.
+        # It returns `nil` if no record matches the condition.
+        #
+        # Example:
+        # ```
+        # tag = Tag.all.get("name=?", "crystal")
+        # ```
+        def get(raw_predicate : String, *args)
+          get(raw_predicate, args.to_a)
+        end
+
+        # Returns the model instance matching the given raw SQL condition with positional parameters.
+        #
+        # This method allows retrieving a record based on a custom SQL condition using an array of parameters.
+        # It returns `nil` if no record matches the condition.
+        #
+        # Example:
+        # ```
+        # tag = Tag.all.get("name=? AND is_active=?", ["crystal", true])
+        # ```
+        def get(raw_predicate : String, params : Array)
+          raise_empty_raw_predicate if raw_predicate.empty?
+
+          raw_params = [] of ::DB::Any
+          raw_params += params
+
+          get(Node.new(raw_predicate: raw_predicate, params: raw_params))
+        end
+
+        # Returns the model instance matching the given raw SQL condition with named parameters.
+        #
+        # This method allows retrieving a record based on a custom SQL condition using named parameters.
+        # It returns `nil` if no record matches the condition.
+        #
+        # Example:
+        # ```
+        # tag = Tag.all.get("name=:name AND is_active=:active", name: "crystal", active: true)
+        # ```
+        def get(raw_predicate : String, **kwargs)
+          get(raw_predicate, kwargs.to_h)
+        end
+
+        # Returns the model instance matching the given raw SQL condition with a named parameters hash.
+        #
+        # This method allows retrieving a record based on a custom SQL condition using a hash of named parameters.
+        # It returns `nil` if no record matches the condition.
+        #
+        # Example:
+        # ```
+        # tag = Tag.all.get("name=:name", {name: "crystal"})
+        # ```
+        def get(raw_predicate : String, params : Hash | NamedTuple)
+          raise_empty_raw_predicate if raw_predicate.empty?
+
+          raw_params = {} of String => ::DB::Any
+          params.each { |k, v| raw_params[k.to_s] = v }
+
+          get(Node.new(raw_predicate: raw_predicate, params: raw_params))
+        end
```

Comment:
Same ask here: can we add specs for this specific variation of the method (with both hashes and named tuples)?